src/convert.rs

```diff
@@ -494,11 +532,41 @@ pub fn elf_to_tbf<W: Write>(
         }
     }
 
-    // Add the relocation data to our total length. Also include the 4 bytes for
-    // the relocation data length.
+    // Add the relocation data to our total length. Also include the
+    // 4 bytes for the relocation data length.
     binary_index += relocation_binary.len() + mem::size_of::<u32>();
+    
+    if program {
```

Comment:
This check should be if any footers are requested.

Reply:
Please see above -- a footer automatically adds a program header, but there are cases when you want a program header but do not have footers.

Reply:
Removed.